src/ensembl/src/shared/components/pointer-box/pointer-box-inline-styles.ts

```diff
@@ -0,0 +1,263 @@
+        }
+      };
+    case Position.RIGHT_BOTTOM:
+      return {
+        bodyStyles: {
+          transform: `translateY(-${pointerOffset + halfPointerWidth}px)`
+        },
+        pointerStyles: {
+          left: `4px`,
+          top: `${pointerOffset}px`,
+          transform: `rotate(-90deg)`,
+          transformOrigin: 'left bottom'
+          // top: `calc(${pointerOffset}px + ${halfPointerWidth}px)`,
+          // transform: `rotate(-90deg) translateY(-${pointerHeight}px)`,
+          // transformOrigin: 'left 0'
+        }
+      };
+  }
+};
+
+const getBodyStylesForRenderingIntoBody = (
+  params: Params
+): { [key: string]: string } => {
+  const { anchor, pointerWidth, pointerHeight, pointerOffset } = params;
+  const anchorBoundingRect = anchor.getBoundingClientRect();
+  const anchorWidth = anchorBoundingRect.width;
+  const halfAnchorWidth = anchorWidth / 2;
+  const anchorHeight = anchorBoundingRect.height;
+  const halfAnchorHeight = anchorHeight / 2;
+  const anchorLeft = Math.round(anchorBoundingRect.left);
+  const anchorTop = Math.round(anchorBoundingRect.top);
+  const halfPointerWidth = Math.floor(pointerWidth / 2);
+
+  switch (params.position) {
+    case Position.TOP_LEFT:
+      return {
+        left: `${anchorLeft + anchorWidth / 2}px`,
+        bottom: `${window.innerHeight -
+          window.scrollY -
+          anchorTop +
+          pointerHeight}px`
+      };
+    case Position.TOP_RIGHT:
+      return {
+        left: `${anchorLeft +
+          halfAnchorWidth -
+          pointerOffset -
+          halfPointerWidth}px`,
+        bottom: `${window.innerHeight -
+          window.scrollY -
+          anchorTop +
+          pointerHeight}px`
+      };
+    case Position.BOTTOM_LEFT:
+      return {
+        top: `${anchorTop + window.scrollY + anchorHeight + pointerHeight}px`,
+        left: `${anchorLeft + anchorWidth / 2}px`
+      };
+    case Position.BOTTOM_RIGHT:
+      return {
+        left: `${anchorLeft +
+          halfAnchorWidth -
+          pointerOffset -
+          halfPointerWidth}px`,
+        top: `${anchorTop + window.scrollY + anchorHeight + pointerHeight}px`
+      };
+    case Position.LEFT_TOP:
+      return {
+        left: `${anchorLeft - pointerHeight}px`,
+        top: `${anchorTop + window.scrollY + halfAnchorHeight}px`
+      };
+    case Position.LEFT_BOTTOM:
+      return {
+        left: `${anchorLeft - pointerHeight}px`,
+        top: `${anchorTop + window.scrollY + halfAnchorHeight}px`
+      };
+    case Position.RIGHT_TOP:
+      return {
+        left: `${anchorLeft + anchorWidth + pointerHeight}px`,
+        top: `${anchorTop +
+          window.scrollY +
+          halfAnchorHeight +
+          pointerOffset}px`
+      };
+    case Position.RIGHT_BOTTOM:
+      return {
+        left: `${anchorLeft + anchorWidth + pointerHeight}px`,
+        top: `${anchorTop + window.scrollY + halfAnchorHeight}px`
+      };
+  }
+};
+
+const getBodyStylesForRenderingIntoAnchor = (
```

Comment:
Is this for the document body or something else?

Reply:
For pointerBox body.
The pointerBox consists of two parts — one is the rectangle, the other the pointing triangle.
```
_______
|       \
|       /
|______|
```
I called the rectangle the body of the pointerBox and the triangle the pointer.

Reply:
This is slightly confusing as we also have `getBodyStylesForRenderingIntoBody`. Can it be changed to something like `Wrapper` or `Container`?

Reply:
I guess we can call this part of the PointerBox box rather than body, if that would help:

![image](https://user-images.githubusercontent.com/6834224/77652114-aba02280-6f65-11ea-8475-6dd3c2e03a45.png)

Reply:
Yes, that would work too.

Reply:
✔️